src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1131,8 +1134,12 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::DoInit(otInstanc
     memset(&mSrpClient, 0, sizeof(mSrpClient));
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
 
+#ifndef _NO_NETWORK_COMMISSIONING_DRIVER_
+    InterfaceEnabled = sGenericThreadDriver.GetEnabled();
```

Comment:
It introduces a cyclic dependency. Normally, it should be Thread driver talking to `ThreadStackManager` and not the other way around.

But overall, why does this have to involve the thread driver and persistent storage. Wouldn't it be enough to provide some `CHIP_DEVICE_CONFIG_ENABLE_IF_AUTOSTART` configuration which you could disable in your app and only enable the Thread interface if needed?

Reply:
The cyclic dependency is indeed a problem. But Thread interface may be enabled and disabled during program running, the compiling configuration may not fit the requires. 

For example, if a device is commissioned successfully through Thread, it should auto-start Thread network. But if commissioning failed, we want to disable Thread interface and try other network interface like Wi-Fi. In this case, the Thread should not auto-start.

Reply:
The compiling configuration for disabling the autostart should allow you to start or stop the interface based on your application's requirements. On top of that you can implement persisting the interface state and either call `SetThreadEnable()` right after `InitThreadStack()` or not. Wouldn't that be convenient for you?

My point is that this PR seems to implement a very specific use case (e.g. the interface is automatically brought up if `ConnectNetwork` is called) and I wonder if we can keep the common code as generic as possible.

Reply:
Thanks, great that you got rid of the cyclic dependency. I'm not sure if I understand precisely how `InterfaceEnabled` should actually work but the logic makes sense to me.